frontend/src/metabase/components/Timeline.jsx

```diff
@@ -0,0 +1,95 @@
+import React, { useMemo } from "react";
+import PropTypes from "prop-types";
+import cx from "classnames";
+import _ from "underscore";
+import styled from "styled-components";
+import moment from "moment";
+
+import { color } from "metabase/lib/colors";
+
+import Icon from "metabase/components/Icon";
+
+const TimelineContainer = styled.div`
+  position: relative;
+  margin-left: ${props => props.leftShift}px;
+  margin-bottom: ${props => props.bottomShift}px;
+`;
+
+const TimelineItem = styled.div`
+  transform: translateX(-${props => props.leftShift}px);
+  white-space: pre-line;
+`;
+
+// shift the border down slightly so that it doesn't appear above the top-most icon
+const Border = styled.div`
+  position: absolute;
+  top: ${props => props.borderShift}px;
+  left: 0;
+  right: 0;
+  bottom: -${props => props.borderShift}px;
+  border-left: 1px solid ${color("border")};
+`;
+
+const Timeline = ({ className, items = [], renderFooter }) => {
+  const iconSize = 20;
+  const halfIconSize = iconSize / 2;
+
+  const sortedFormattedItems = useMemo(() => {
```

Comment:
Should the timeline itself be concerned with this sorting?

Reply:
I debated this before adding it and settled on "yes" because you'll only ever want to use `Timeline` in a situation where you have ordered events, ascending (which we can add later) or descending, and it doesn't make much sense to offer a use case where the events are out of order. Because of that, you'd see every implementation of `<Timeline items={item} />` require a `items = items.sort(...)` somewhere.

The major downside is you remove control over whether this is ascending or descending, but I figured adding a custom `sort` prop or just a simple `ascending` prop or something would be a reasonable add.

Reply:
Makes sense!